packages/entity-server/src/routes/hierarchy/middleware/attachEntityContext.ts

```diff
@@ -36,30 +58,11 @@ export const attachSingleEntityContext = async (
   try {
     const { entityCode } = req.params;
 
-    const entity = await req.models.entity.findOne({ code: entityCode });
-    if (!entity) {
-      throwNoAccessError(entityCode);
-    }
-
-    const { hierarchyId } = req.ctx;
-    // Root type shouldn't be locked into being a project entity, see: https://github.com/beyondessential/tupaia-backlog/issues/2570
-    const rootEntity = await entity.getAncestorOfType(hierarchyId, 'project');
-
-    const allowedCountries = await getAllowedCountries(req, rootEntity);
-
-    if (allowedCountries.length < 1) {
-      throwNoAccessError(entityCode);
-    }
-
-    if (!userCanAccessEntity(entity, allowedCountries)) {
-      throwNoAccessError(entityCode);
-    }
-
-    req.ctx.entity = entity;
-    req.ctx.allowedCountries = allowedCountries;
+    const context = await validateEntitiesAndBuildContext(req, [entityCode]);
 
-    const { filter } = req.query;
-    req.ctx.filter = extractFilterFromQuery(allowedCountries, filter);
+    [req.ctx.entity] = context.entities;
```

Comment:
I didn't know you could use array destructuring without a `const` or `let` - nice!

Reply:
I actually didn't know this either, but our linter was yelling at me to `use array destructuring` and I had to keep trying things until it was happy 😝

Reply:
Ah, the benevolent dictatorship of linters... 😆